applications/vanilla/controllers/api/DiscussionsApiController.php

```diff
@@ -575,9 +575,12 @@ public function patch($id, array $body) {
             $categoryID = $discussionData['CategoryID'];
         }
 
-        $this->fieldPermission($body, 'closed', 'Vanilla.Discussions.Close', $categoryID);
-        $this->fieldPermission($body, 'pinned', 'Vanilla.Discussions.Announce', $categoryID);
-        $this->fieldPermission($body, 'sink', 'Vanilla.Discussions.Sink', $categoryID);
+        $category = CategoryModel::categories($categoryID);
+        $permissionCategoryID = $category->PermissionCategoryID;
```

Comment:
isn't category an array? you're accessing it as an object.

also you want to check if you have a category before doing
 $permissionCategoryID = $category->PermissionCategoryID;
something like 

 if ($category) {
                $permissionCategoryID = ...
            } else {
                $permissionCategoryID = -1;
            }